File: StudyBuddy.Model/Model/TagList.cs
using System.Collections.Generic;
using System.Linq;

namespace StudyBuddy.Model
{
    public class TagList
    {
        public int Count { get; set; }
        public IEnumerable<Tag> Objects { get; set; }

        public TagList()
        {
        }

        public TagList(IEnumerable<Tag> objects) : this(objects, objects.Count())
        {
        }

        public TagList(IEnumerable<Tag> objects, int count)
        {
            Objects = objects;
            Count = count;
        }

        public static string ToString(IEnumerable<Tag> objects)
        {
            return string.Join(" ", objects.Select(obj => "#" + obj.Name.ToLower()));
        }
    }
}

File: StudyBuddy.Backend/src/app/model/taglist.ts
import { Tag } from "./tag";

export class TagList {
    count: number = 0;
    objects: Tag[] = [];

    constructor(result) {
        this.count = result['count'];

        for (let index in result['objects']) {
            this.objects.push(Tag.fromApi(result['objects'][index]));
        }
    }
}